src/gtest/test_keys.cpp

```diff
@@ -27,6 +27,20 @@ TEST(Keys, EncodeAndDecodeSapling)
             auto sk2 = boost::get<libzcash::SaplingExtendedSpendingKey>(spendingkey2);
             EXPECT_EQ(sk, sk2);
         }
+        {
+            auto extfvk = sk.ToXFVK();
+            std::string vk_string = EncodeViewingKey(extfvk);
+            EXPECT_EQ(
+                vk_string.substr(0, 7),
+                Params().Bech32HRP(CChainParams::SAPLING_EXTENDED_FVK));
+
+            auto viewingkey2 = DecodeViewingKey(vk_string);
```

Comment:
A bit unrelated and maybe a dumb question but what is the `2` in `viewingkey2` is trying to say ? I understand this was done by following other variables in the test just as `spendingkey2` and `paymentaddr2` however i am failing to see the purpose.

Reply:
Simply a quick way to indicate that it is the decoded key, not the original.